include/nbl/asset/IFramebuffer.h

```diff
@@ -75,18 +77,24 @@ class IFramebuffer
             {
                 const auto& aParams = a->getCreationParameters();
 
+                if (aParams.image->getCreationParameters().extent.width < params.width)
+                    return false;
                 if (aParams.image->getCreationParameters().extent.height < params.height)
                     return false;
                 if (aParams.subresourceRange.layerCount < params.layers)
                     return false;
-                if (aParams.viewType == IGPUImageView::ET_3D)
+                if (aParams.subresourceRange.levelCount != 1u)
+                    return false;
+                if (aParams.viewType == ImageViewType::ET_3D)
```

Comment:
you can attach a 3D texture to a framebuffer (as a layer or as a layered fframebuffer)

Reply:
@Crisspl useful to think about when touching up your FBO cache